notebooks/week1/hw1.jl

```diff
@@ -0,0 +1,1418 @@
+        el
+    end
+end
+
+# ╔═╡ 29dfe3d6-c353-4081-8192-b12f374bf702
+filter!(LOAD_PATH) do path
+	path != "@v#.#"
+end;
+
+# ╔═╡ 65780f00-ed6b-11ea-1ecf-8b35523a7ac0
+begin
+	import Pkg
+	Pkg.activate(mktempdir())
+	Pkg.add([
+			Pkg.PackageSpec(name="Images", version="0.22.4"), 
+			Pkg.PackageSpec(name="ImageMagick", version="0.7"), 
+			Pkg.PackageSpec(name="PlutoUI", version="0.7"), 
+			Pkg.PackageSpec(name="HypertextLiteral", version="0.5")
+			])
+
+	using Images
+	using PlutoUI
+	using HypertextLiteral
+end
+
+# ╔═╡ ac8ff080-ed61-11ea-3650-d9df06123e1f
+md"""
+
+# **Homework 1** - _images and arrays_
+`18.S191`, Spring 2021
+
+This notebook contains _built-in, live answer checks_! In some exercises you will see a coloured box, which runs a test case on your code, and provides feedback based on the result. Simply edit the code, run it, and the check runs again.
+
+_For MIT students:_ there will also be some additional (secret) test cases that will be run as part of the grading process, and we will look at your notebook and write comments.
+
+Feel free to ask questions!
+"""
+
+# ╔═╡ 911ccbce-ed68-11ea-3606-0384e7580d7c
+# edit the code below to set your name and kerberos ID (i.e. email without @mit.edu)
+
+student = (name = "Jazzy Doe", kerberos_id = "jazz")
+
+# press the ▶ button in the bottom right of this cell to run your edits
+# or use Shift+Enter
+
+# you might need to wait until all other cells in this notebook have completed running. 
+# scroll down the page to see what's up
+
+# ╔═╡ 8ef13896-ed68-11ea-160b-3550eeabbd7d
+md"""
+
+Submission by: **_$(student.name)_** ($(student.kerberos_id)@mit.edu)
+"""
+
+# ╔═╡ 5f95e01a-ee0a-11ea-030c-9dba276aba92
+md"""
+#### Intializing packages
+
+_When running this notebook for the first time, this could take up to 15 minutes. Hang in there!_
+"""
+
+# ╔═╡ 540ccfcc-ee0a-11ea-15dc-4f8120063397
+md"""
+## **Exercise 1** - _Manipulating vectors (1D images)_
+
+A `Vector` is a 1D array. We can think of that as a 1D image.
+
+"""
+
+# ╔═╡ 467856dc-eded-11ea-0f83-13d939021ef3
+example_vector = [0.5, 0.4, 0.3, 0.2, 0.1, 0.0, 0.7, 0.0, 0.7, 0.9]
+
+# ╔═╡ ad6a33b0-eded-11ea-324c-cfabfd658b56
+md"""
+$(html"<br>")
+#### Exerise 1.1
+👉 Make a random vector `random_vect` of length 10 using the `rand` function.
+"""
+
+# ╔═╡ f51333a6-eded-11ea-34e6-bfbb3a69bcb0
+random_vect = missing # replace `missing` with your code!
+
+# ╔═╡ 5da8cbe8-eded-11ea-2e43-c5b7cc71e133
+begin
+	colored_line(x::Vector) = hcat(Gray.(Float64.(x)))'
+	colored_line(x::Any) = nothing
+end
+
+# ╔═╡ 56ced344-eded-11ea-3e81-3936e9ad5777
+colored_line(example_vector)
+
+# ╔═╡ b18e2c54-edf1-11ea-0cbf-85946d64b6a2
+colored_line(random_vect)
+
+# ╔═╡ c5aed290-f8a0-4246-9e66-b1d20a5d3465
+#= md"#### Exerise 1.2
+The function `colored_line` displays a vector of numbers as a sequence of colors.
+
+👉 Create a black line Make a function `my_sum` using a `for` loop, which computes the total of a vector of numbers." =#
```

Comment:
I think the instructions for making a function are copied from below on accident? Though this all seems commented out so maybe just delete this cell?